README.md

```diff
@@ -49,34 +48,22 @@ Once the mutation succeeds, we could request apollo to send a refetch request to
-
-- This approach only works for entities that don't have nested entities and not needed in multiple interfaces using different queries.
-
-- This approach is usually not practical for top level entities that has deep nested entities. Eg. Social network application has categorised topics, each topic will have posts, each post will have comments, etc. Now while creating a topic, it's not practical to fetch the entire nested entities in create mutation response.
-
-**Partial payload:** This is the most practical approach and suits for most cases. Apollo is smart enough to refetch when there are entities with missing fields in cache for an active query. Eg. in an address book, if an address query needs 5 fields and if we insert a new address into cache with only 1 field, apollo identifies these missing fields and sends a refetch request for the address query to get latest data. We could make use of this mechanism for most create experiences. We only need to request id (or a few more relevant fields) in create mutations and insert that into where ever it needs to be inserted. We then let apollo take care of fetching whichever query is active and if that query is missing a required field. This way we will be able to build cache and mutation solutions independent of feature specific requirements and reuse these solutions across the app in different interfaces.
-
-- Very practical. Always resort to this approach or at least try to.
+Sometimes it might not be practical to return all fields changed by a mutation. Usually this happens when a mutation affects many different unrelated entities. For example, if a todo's due date is changed in a project management app, it might also change the due date of many other projects that it is related to. In this case you might either need to write your own custom update logic in the FE using apollo mutation's `update` function, refetch the other affected queries, or alternatively create your own custom mutation response that includes all updated entities.
 
 ## APIs for cache update
 
 There are two ways to insert data into apollo's cache after mutation response or at any time using apollo's cache API.
 
 `cache.updateFragment`: This is by far the easiest way to insert data into apollo cache. This cache API is very similar to how you write actual graphql queries/mutation. The only difference is that the updateFragment is a shorthand for `readFragment` + `writeFragment` together and instead of making a network request to read/write, apollo reads/writes from the cache. So `cache.updateFragment` is more of an apollo's way to reading and writing data to cache using the familiar graphql query/mutation schema.
 
-- be warned and only use this when you want to completely replace data in cache or only when you want to write object shaped data into cache
```

Comment:
I removed this, because I don't think this is true. You can use updateFragment to update partial data just as easily as with cache.modify?

The main difference in my experience is that (like you say below) cache.modify does not apply merge functions, but updateFragment does.